components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js

```diff
@@ -992,63 +992,133 @@ define(['require', 'log', 'jquery', 'backbone', 'lodash', 'dropElements', 'dagre
          * @function Auto align the diagram
          */
         DesignGrid.prototype.autoAlignElements = function () {
-            //TODO auto aligning does not support 'Partition'
             var self = this;
-            var g = new dagre.graphlib.Graph();
-            g.setGraph({
-                rankdir: 'LR'
-            });
-            g.setDefaultEdgeLabel(function () {
+
+            var graph = new dagre.graphlib.Graph({compound: true});
+            graph.setGraph({rankdir: "LR"});
+            graph.setDefaultEdgeLabel(function () {
```

Comment:
Add a comment stating why this method can't be removed and what is the purpose of it